src/client/routes/BUILD.bazel

```diff
@@ -1,9 +1,17 @@
-load("//bazel/js:defaults.bzl", "build_module", "build_route", "generate_route_manifest", "pkg_web")
+load("@npm//webpack-cli:index.bzl", "webpack_cli")
+load("//bazel/js:defaults.bzl", "build_module", "pkg_web")
+load("@rules_spa//spa:defs.bzl", "build_route", "generate_route_manifest")
 
 build_route(
     name = "default",
-    data = ["default.tsx"],
+    srcs = ["default.tsx"],
+    data = [
+        "//:package.json",
+        "@npm//:node_modules",
```

Comment:
Is there any way the route can infer pulling in the top-level package.json and :node_modules instead of these included as data here? At least the former doesn’t seem like it would be directly clear, though I see why it is declared here. I wonder if a consumer of this SPA tooling would realistically know to include these, aside from the examples we share.

Reply:
There is no way for the macro to know what the npm workspace name is without you specifying it. You also aren't guaranteed to have a top level package.json depending on the set up/migration in progress. It's not ideal, I agree, but I don't see a faster more efficient way to roll out yet.